tests/plugins/test_add_filesystem.py

```diff
@@ -101,7 +101,7 @@ def _mockBuildImageOz(*args, **kwargs):
             .replace_with(get_task_result_mock).once())
 
     session.should_receive('listTaskOutput').and_return([
-        'fedora-23-1.0.tar.gz',
+        'fedora-23-1.0.x86_64.tar.gz',
```

Comment:
This particular bit of code seems rather tied up with several of the self-tests, but the additions pushed for test_build_filesystem_from_task_id go a ways to make sure we match arch properly too, when it's provided. I think we're good to go here, and will start working on cleaning things up for a formal pull request -- right after some functional testing with @lcarva.